### ISSUE TITLE
How to create different panels for two windows

### ISSUE THREAD

**Issue by pisuto** [stevenlovegrove/Pangolin#435]
Hello,
Sorry for bothering you. Now I have two threads and in each thread I create a new window by pangolin::CreateWindowAndBind. But when I want to create different panels for each window using pangolin::CreatePanel, I meet a problem that only one window has all the buttons and the other one is null. And if I set the panels the same name, both of windows have the all buttons which aren't separated. So how can I solve this problem? Thank you very much.
one panel:
`pangolin::Var<bool> menu_drawlines("menu1.Draw Lines", false, true);                                     pangolin::Var<bool> menu_drawpolygons("menu1.Draw Polygons", false, true);`
the other panel:
`pangolin::Var<bool> menuShowPoints("menu2.Show Points", false,true);`

**Comment by stevenlovegrove** [comment]
Have you set the correct prefix when creating the second panel?

**Comment by pisuto** [comment]
yes,
thread one:
` pangolin::CreateWindowAndBind("Viewer1",1024,768);
pangolin::CreatePanel("menu1").SetBounds(0.0,1.0,0.0,pangolin::Attach::Pix(175));`
and thread two:
`pangolin::CreateWindowAndBind("Viewer2",640,480);
pangolin::CreatePanel("menu2").SetBounds(0.0,1.0,0.0,pangolin::Attach::Pix(200));`

**Comment by pisuto** [comment]
and I used .NumChildren() function and found the num of children was zero. But I am sure that the name of panel is correct...

**Comment by pisuto** [comment]
ok, I solved my problem...

when I set one panel name "### **menu",** the another panel will attach to this panel...
and I set it another name, this problem disappears.
So I think maybe the string  "### **menu"** is the problem. But I don't know why.

**Comment by NikolausDemmel** [comment]
I had a similar problem when one panel name is a prefix of the other, i.e. `menu` and `menu2` or similar. The system simply checks if a varaible name starts with the panel name, and then adds it. 

Therefore, if you add a variable with name `menu2.foobar`, it should be added to both panels with name `menu` and `menu2`.

https://github.com/stevenlovegrove/Pangolin/blob/d9daba62dfa3f6c786699723621b5ca4566206dd/src/display/widgets/widgets.cpp#L142

https://github.com/stevenlovegrove/Pangolin/blob/d9daba62dfa3f6c786699723621b5ca4566206dd/include/pangolin/var/varstate.h#L77

**Comment by pisuto** [comment]
> I had a similar problem when one panel name is a prefix of the other, i.e. `menu` and `menu2` or similar. The system simply checks if a varaible name starts with the panel name, and then adds it.
> 
> Therefore, if you add a variable with name `menu2.foobar`, it should be added to both panels with name `menu` and `menu2`.
> 
> https://github.com/stevenlovegrove/Pangolin/blob/d9daba62dfa3f6c786699723621b5ca4566206dd/src/display/widgets/widgets.cpp#L142
> 
> https://github.com/stevenlovegrove/Pangolin/blob/d9daba62dfa3f6c786699723621b5ca4566206dd/include/pangolin/var/varstate.h#L77

Yeah, thank you. now I understand what Steven said. I think it needs to be fixed. Otherwise, it maybe a problem for newcomer.

**Comment by pisuto** [comment]
Yeah, thank you very muchl. Now I understand what Steven said. I think it needs to be fixed. Otherwise, it maybe a problem for newcomer.
